src/java.base/share/classes/sun/nio/ch/DatagramChannelImpl.java

```diff
@@ -1589,17 +1483,13 @@ private void repairSocket(InetSocketAddress target)
     /**
      * Defines static methods to access AbstractSelectableChannel non-public members.
      */
-    @SuppressWarnings("removal")
     private static class AbstractSelectableChannels {
         private static final Method FOREACH;
         static {
             try {
-                PrivilegedExceptionAction<Method> pae = () -> {
-                    Method m = AbstractSelectableChannel.class.getDeclaredMethod("forEach", Consumer.class);
-                    m.setAccessible(true);
-                    return m;
-                };
-                FOREACH = AccessController.doPrivileged(pae);
+                Method m = AbstractSelectableChannel.class.getDeclaredMethod("forEach", Consumer.class);
+                m.setAccessible(true);
+                FOREACH = m;
```

Comment:
Should we use this opportunity to use MethodHandle instead?

Reply:
It could be changed but not related to this PR so I'd prefer not do it here.